pynumaflow/function/server.py

```diff
@@ -0,0 +1,92 @@
+import asyncio
+import logging
+from os import environ
+
+from pynumaflow.function import udfunction_pb2, Datum
+from pynumaflow.function import udfunction_pb2_grpc
+
+from google.protobuf import timestamp_pb2 as _timestamp_pb2
+
+import grpc
+from typing import Callable, Any
+
+from pynumaflow._constants import (
+    FUNCTION_SOCK_PATH,
+    DATUM_KEY,
+)
+from pynumaflow.function import Messages
+
+if environ.get("PYTHONDEBUG"):
+    logging.basicConfig(level=logging.DEBUG)
+
+_LOGGER = logging.getLogger(__name__)
+
+UDFMapCallable = Callable[[str, Datum, Any], Messages]
+
+
+class UserDefinedFunctionServicer(udfunction_pb2_grpc.UserDefinedFunctionServicer):
+    def __init__(self, map_handler: UDFMapCallable, sock_path=FUNCTION_SOCK_PATH):
+        self.__map_handler: UDFMapCallable = map_handler
+        self.sock_path = sock_path
+        self._cleanup_coroutines = []
+
+    def MapFn(self, request: udfunction_pb2.Datum, context):
```

Comment:
Python follows snake case for variable and function names

```suggestion
    def map(self, request: udfunction_pb2.Datum, context):
```
Add typing for `context`

Reply:
added a new type as discussed offline